### PR TITLE
Feat/met 2604 signal for raw xml availability

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package eu.europeana.metis.core.rest.execution.details;
+
+import eu.europeana.cloud.common.model.dps.TaskState;
+import eu.europeana.metis.core.workflow.plugins.ExecutionProgress;
+
+public class PluginProgressView {
```
Comment (stzanakis) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java:6]: I think we can have this class declared inside the PluginView?
What do you think?

Comment (stzanakis) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java:6]: Add javadoc

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  private final Date createdDate;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date startedDate;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date updatedDate;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date finishedDate;
+  private final List<PluginView> metisPlugins;
+
+  public ExecutionView(WorkflowExecution execution,
```
Comment (stzanakis) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java:33]: Add javadoc

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import eu.europeana.metis.CommonStringValues;
+import eu.europeana.metis.core.workflow.WorkflowExecution;
+import eu.europeana.metis.core.workflow.WorkflowStatus;
+import eu.europeana.metis.core.workflow.plugins.AbstractMetisPlugin;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class ExecutionView {
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java:13]: Add javadoc

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import com.fasterxml.jackson.annotation.JsonFormat;
+import eu.europeana.metis.CommonStringValues;
+import eu.europeana.metis.core.workflow.plugins.AbstractExecutablePlugin;
+import eu.europeana.metis.core.workflow.plugins.AbstractMetisPlugin;
+import eu.europeana.metis.core.workflow.plugins.DataStatus;
+import eu.europeana.metis.core.workflow.plugins.PluginStatus;
+import eu.europeana.metis.core.workflow.plugins.PluginType;
+import java.util.Date;
+
+public class PluginView {
```
Comment (stzanakis) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java:12]: Add javadoc

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  private final DataStatus dataStatus;
+  private final String failMessage;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date startedDate;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date updatedDate;
+  @JsonFormat(pattern = CommonStringValues.DATE_FORMAT)
+  private final Date finishedDate;
+  private final String externalTaskId;
+  private final PluginProgressView executionProgress;
```
Comment (stzanakis) [inline metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java:26]: I think we should rename this too to match the type name, what do you think?

**Review by stzanakis — CHANGES_REQUESTED** [review]

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jochen-vermeulen) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/ExecutionView.java:13]: I was trying to be consistent with the classes in the overview package instead. But this is fine with me.

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java:6]: In the overview package the equivalent class is also separated in its own file. Perhaps just keep it consistent with that?

Comment (jochen-vermeulen) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java:6]: I guess I forgot to add Javadoc in those new files. I'll add it.

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java (hunk 5 above)
Comment (jochen-vermeulen) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java:26]: The problem is that this then requires frontend changes... I was trying to keep everything as much the same as possible for the frontend.

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java (hunk 1 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginProgressView.java:9]: Then fine for now but I think we need to rethink the names of those two classes..
Having the same name in two different packages can be confusing.

File: metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java (hunk 5 above)
Comment (stzanakis) [reply metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/rest/execution/details/PluginView.java:30]: Then fine for now but I think we need to rethink the names of those two classes..
Having the same name in two different packages can be confusing.

**Pull request merged**